### PR TITLE
Fixed CMS deprectation warnings in RecoMuon/TrackingTools

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- changed to thread-friendly module types
- added esConsumes calls

#### PR validation:

Code compiles without generating CMS deprecation warnings.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h [hunk 1]
```diff
@@ -95,11 +95,12 @@ class MuonErrorMatrixAdjuster : public edm::EDProducer {
   bool theRescale;
 
   /// holds the error matrix parametrization
-  edm::ParameterSet theMatrixProvider_pset;
-  MuonErrorMatrix* theMatrixProvider;
+  std::unique_ptr<MuonErrorMatrix> theMatrixProvider;
 
   /// hold on to the magnetic field
   edm::ESHandle<MagneticField> theField;
```
Comment (clacaputo) [inline RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h:101]: Could this be `const`?

File: RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h [hunk 2]
```diff
@@ -95,11 +95,12 @@ class MuonErrorMatrixAdjuster : public edm::EDProducer {
...
+  edm::ESGetToken<MagneticField, IdealMagneticFieldRecord> theFieldToken;
```
Comment (clacaputo) [inline RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h:102]: as above

File: RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h [hunk 3]
```diff
@@ -95,11 +95,12 @@ class MuonErrorMatrixAdjuster : public edm::EDProducer {
...
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> theHttopoToken;
```
Comment (clacaputo) [inline RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h:103]: As above

File: RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h (hunk 1 above)
Comment (Dr15Jones) [reply RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h:101]: The class MuonErrorMatrix has no `const` methods and therefore making it `const` would not work:
https://github.com/cms-sw/cmssw/blob/master/RecoMuon/TrackingTools/interface/MuonErrorMatrix.h

File: RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h (hunk 3 above)
Comment (Dr15Jones) [reply RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h]: done

**Comment by Dr15Jones** [comment]
please test

File: RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h (hunk 1 above)
Comment (clacaputo) [reply RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h:101]: Hi @Dr15Jones , actually I was referring to `edm::ESHandle<MagneticField> theField`

Comment (clacaputo) [reply RecoMuon/TrackingTools/plugins/MuonErrorMatrixAdjuster.h:101]: ah, thanks for the explanation

**Comment by clacaputo** [comment]
+reconstruction

- technical PR
- no reco changes

**Comment by perrotta** [comment]
+1

**Pull request merged**